### ISSUE TITLE
Use `create or replace view` when creating views in postgres using synchronize

### ISSUE THREAD

**Issue by dncnkrs** [typeorm/typeorm#7966]
## Issue Description

When a view expression is updated while using synchronize, sometimes the previous view is not dropped before typeorm attempts to recreate the view. This causes an error because the view already exists. Tweaking the generated SQL would eliminate this issue.

### Expected Behavior

When a view is created on postgres, the syntax `create or replace view` is used in place of `create view`. If the view name is in conflict, the previous view will be replaced.

### Actual Behavior

`create view` is currently used.

### My Environment

| Dependency          | Version  |
| ---                 | ---      |
| Operating System    | Ubuntu 20.04         |
| Node.js version     | 14.15.4  | 
| Typescript version  | ^4.2.4  | 
| TypeORM version     | ^0.2.32  | 

### Relevant Database Driver(s)

| DB Type              | Reproducible  |
| ---                  | ---           |
| `aurora-data-api`    | no            |
| `aurora-data-api-pg` | no            |
| `better-sqlite3`     | no            |
| `cockroachdb`        | no            |
| `cordova`            | no            |
| `expo`               | no            |
| `mongodb`            | no            |
| `mysql`              | no            |
| `nativescript`       | no            |
| `oracle`             | no            |
| `postgres`           | yes            |
| `react-native`       | no            |
| `sap`                | no            |
| `sqlite`             | no            |
| `sqlite-abstract`    | no            |
| `sqljs`              | no            |
| `sqlserver`          | no            |

**Comment by cooperfrench95** [comment]
Agree, can't see any reason not to use `CREATE OR REPLACE` over `CREATE`. Currently the fact that views (as far as I can tell) are only changed using create or drop statements can cause migrations to fail and require manually dropping both the view itself + the typeorm metadata table before typeorm migrations can be generated correctly again. For example, if you update a `ViewEntity` to add the `materialized: true` it doesn't generate the migration correctly and you need to drop the view first.

**Comment by oxatruchdev** [comment]
are there any updates in this issue? i updated a view and it's trying to drop the view without checking if it has dependencies which is my case so it makes it impossible to run the migration without dropping the dependencies.

And if i modify the migration manually to use create or replace and update the metadata in `typeorm_metadata` and if I create a new migration for something different it tries to create another migration with the create view statement again even tho it's been modified before

**Comment by clintonb** [comment]
Can we re-open this issue? It's burned us in production at least twice because of lock contention (which is a separate issue). A single, atomic, call is cleaner and avoids issues.
